### PR TITLE
Do not start duplicate permission dialogs

### PR DESCRIPTION
**Pull request by zakharvoit**
Update WolvicPermissionManager logic to prevent dialogs for identical permission requests from being started twice. This can happen when an Android activity is restarted due to a denied Android permission. In that case we want to rely on the initial user response instead of starting another user dialog.

### CODE REVIEW

**Review by mshin-wolvic — APPROVED** [review]

File: wolvic/wolvic_permission_manager.cc [hunk 1]
```diff
@@ -269,10 +282,27 @@ void WolvicPermissionManager::OnPermissionResult(
...
+                         in_progress_requests_.end(), [&](const auto& request) {
+                           return request->description.requesting_origin ==
+                                      description.requesting_origin &&
+                                  request->description.permissions ==
+                                      description.permissions;
+                         });
+  if (it == in_progress_requests_.end()) {
+    return nullptr;
+  }
+  return it->get();
```
Comment (svillar) [inline wolvic/wolvic_permission_manager.cc]: maybe using ternary operator to do both returns in a single line?

**Review by svillar — CHANGES_REQUESTED** [review]
LGTM, I just have a minor request

File: wolvic/wolvic_permission_manager.cc (hunk 1 above)
Comment (zakharvoit) [reply wolvic/wolvic_permission_manager.cc]: Fixed.

**Review by svillar — APPROVED** [review]
LGTM

**Pull request merged**